### PR TITLE
Rearchitect CSV import with staged pipeline

### LINKED ISSUE #58: Import by Format - Import from CSV - Doesn't allow Colum Selection  / Matching
When using 'Import by Format - Import from CSV', the Map option allowing mapping Columns to fields is not presented.

e.g. Manually Created CSV.

```
Dive ,Date,Time,Depth,Duration,,,Notes
1,1998-08-04,am,12,48,,,First Dive
2,1998-08-04,pm,12,62,,,
3,1998-08-05,11:22,15,45,,,
4,1998-08-05,pm,15,45,,,
5,1998-08-06,am,28,47,,,
6,1998-08-06,pm,25,57,,,
7,1998-08-07,10:32,26.5,52,,,
8,1998-08-07,12:38,24.2,52,,,
9,1998-08-07,16:28,19.5,46,,,
10,1998-08-08,10:28,20,66,,,
11,1998-08-08,pm,18.9,66,,,
12,1998-08-09,11:05,19.2,57,,,

```
where 
'Dive' would map to 'dive Number'
'Date' to 'date'
'Time' to 'time'
'Depth' to 'max Depth'
(next two columns skipped)
'Notes' to 'notes'

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

- Replaces the dual CSV parser system (legacy `CsvImportService` + `CsvImportParser`) with a single **staged pipeline**: Parse → Detect → Configure → Transform → Correlate → Preview → Import
- Adds **multi-file CSV import** for Subsurface (dive list + dive profile CSVs correlated by dive number)
- Adds **7 built-in presets** with the architecture to trivially add more (data-only definitions)
- Adds **user-saveable presets** (database table + repository for custom field mappings)
- Extracts **all entity types** from CSV: dives, sites, tanks (up to 6 per dive), dive profiles, buddies, tags, gear
- Fixes #58 (column mapping UI missing — single code path), #59 (LF line endings), #61 (informal time defaults), #63 (AM/PM with seconds), #64 (timezone interpretation option)
- Verifies #62 fix (provider invalidation after import)
- Net **-1,890 lines** (legacy code removed)
- **155 new tests** added (3,738 total passing)

## Architecture

```
Parse → Detect → Configure → Transform → Correlate → Preview → Import
  │        │          │           │            │
  │   PresetRegistry  │    TimeResolver    EntityExtractors
  │   (7 built-in +   │    UnitDetector    (Dive, Tank, Profile,
  │    user presets)   │    ValueConverter   Site, Buddy, Tag, Gear)
  │                    │
  CsvParser        ImportConfiguration
  (line ending     (mappings, time interpretation,
   normalization)   unit settings, entity types)
```

## Test plan

- [x] All 3,738 tests pass (155 new)
- [x] `flutter analyze` reports 0 issues
- [x] Integration tests with real Subsurface CSV exports (24 dives, 16,503 profile samples)
- [x] Multi-tank extraction verified (6-tank support)
- [x] GPS coordinate parsing verified
- [x] Buddy leading-comma edge case handled
- [x] UTC wall-time preservation verified (times not shifted)
- [x] Profile-to-dive correlation verified
- [x] Manual test: import Subsurface CSV through the wizard UI
- [x] Manual test: import unknown CSV through full mapper path
- [x] Manual test: save and reuse a custom preset

Closes #58, #59, #61, #63, #64

### CODE REVIEW

File: lib/features/universal_import/data/csv/presets/csv_preset.dart [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+  static ColumnMapping _columnMappingFromJson(Map<String, dynamic> data) {
+    final transformName = data['transform'] as String?;
+    final ValueTransform? transform = transformName != null
+        ? ValueTransform.values.firstWhere(
+            (e) => e.name == transformName,
+            orElse: () =>
+                throw FormatException('Unknown ValueTransform: $transformName'),
+          )
+        : null;
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/universal_import/data/csv/presets/csv_preset.dart]: Fixed. Unknown transform names are now silently treated as `null` instead of throwing `FormatException`. The preset loads normally with the column unmapped, so users can re-select the transform.

File: lib/features/universal_import/data/models/import_enums.dart [hunk 2]
```diff
@@ -81,6 +81,130 @@ enum SourceApp {
...
+    required this.displayName,
+  });
+
+  /// All supported (app, format) combinations for the override dropdown.
+  static const List<SourceOverrideOption> supported = [
+    SourceOverrideOption(
+      sourceApp: SourceApp.submersion,
+      format: ImportFormat.csv,
+      displayName: 'Submersion (CSV)',
+    ),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/universal_import/data/models/import_enums.dart]: Every enum in the codebase uses `displayName` getters in English. Localizing enum display names is a separate effort — not in scope for the CSV import rearchitect.

File: lib/features/universal_import/data/csv/presets/csv_preset.dart [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        entry.value as Map<String, dynamic>,
+      );
+    }
+
+    final rawEntities =
+        (data['supportedEntities'] as List<dynamic>?)?.cast<String>() ?? [];
+    final supportedEntities = rawEntities
+        .map(
+          (name) =>
+              ImportEntityType.values.where((e) => e.name == name).firstOrNull,
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/universal_import/data/csv/presets/csv_preset.dart]: `firstOrNull` is Dart 3.0 core (`Iterable` extension). No import needed. Compiles and analyzes clean. This has been verified and replied to in three prior rounds.
